### PR TITLE
webgl support 

### PR DESCRIPTION
**Pull request by AlokKumar-CT**
analytics support
user properties suppor

### CODE REVIEW

**Comment by AlokKumar-CT** [comment]
Fix https://wizrocket.atlassian.net/browse/MC-1813

**Comment by AlokKumar-CT** [comment]
MC-1813 release

**Comment by AlokKumar-CT** [comment]
MC-1813 release

File: CleverTap/Runtime/Native/UnityNativeWrapper/UnityNativePreferenceManager.cs
Comment (nzagorchev) [inline CleverTap/Runtime/Native/UnityNativeWrapper/UnityNativePreferenceManager.cs:1]: Is this Preference manager used? Only usage is initialization in UnityNativeEventManager.

File: CleverTap/Runtime/Native/UnityNativeWrapper/UnityNativeSessionManager.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        internal int GetScreenCount() {
+            // TODO: Implement if needed
+            return 1;
+        }
+
+        internal long GetLastSessionLength() {
+            return DateTimeOffset.UtcNow.ToUnixTimeSeconds() - _currentSession.StartTimestamp;
+        }
+
+        internal void UpdateSessionUserIdentity(string userIdentity) {
```
Comment (nzagorchev) [inline CleverTap/Runtime/Native/UnityNativeWrapper/UnityNativeSessionManager.cs:41]: This is not referenced anywhere. Is the user set in the session somehow?

File: CleverTap/Runtime/Native/UnityNativeWrapper/Validators/UnityNativeEventValidator.cs [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+                return false;
+            }
+                
+            if (cleverTapId.Length > 64) {
+                // Log? -> ex. Custom CleverTap ID is greater than 64 characters
+                return false;
+
+            }
+
+            // TODO : Check if this work properly
```
Comment (nzagorchev) [inline CleverTap/Runtime/Native/UnityNativeWrapper/Validators/UnityNativeEventValidator.cs:157]: Is this checked?

File: CleverTap/Runtime/Native/UnityNativeWrapper/Validators/UnityNativeEventValidator.cs [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+            }
+
+            return new UnityNativeValidationResult();
+        }
+
+        internal bool IsValidCleverTapId(string cleverTapId) {
+            // This method is used to check if user custom clverTapId is valid
+            
+            if (string.IsNullOrEmpty(cleverTapId)) {
+                // Log? -> ex. CleverTapUseCustomId has been specified to true in config but custom CleverTap ID is null or empty.
```
Comment (nzagorchev) [inline CleverTap/Runtime/Native/UnityNativeWrapper/Validators/UnityNativeEventValidator.cs:147]: Log

File: CleverTap/Runtime/Native/UnityNativeWrapper/UnityNativeWrapper.cs
Comment (nzagorchev) [inline CleverTap/Runtime/Native/UnityNativeWrapper/UnityNativeWrapper.cs:1]: Why is this called Wrapper?

**Comment by nzagorchev** [comment]
Additional comments:
- All events are saved in one queue and save in one batch. Raised events and Profile events must be in separate queues and sent in separate batches.

- `OnUserLogin` does _not_ check the current user. If a new user logs in, a _new_ deviceId _must_ be generated (currently it stays the same). The session _must_ be reset and queues flushed.

- If `LaunchWithCredentials` fails then `UnityNativeAccountManager.Instance.AccountInfo` is not set. Doing a `OnUserLogin` after that results in a Null Pointer exception.

- If a `Date` object is recorded as a profile property but it does not match the formats in the validator (due to locale settings), the validator returns an error. `Date` objects should be accepted.
```The provided value is not a valid date and time.```

- Recording an event results in the following exception (running the app in the Unity Editor)
```
NotSupportedException: Cannot delete UnityNativeEventDBEntry: it has no PK
SQLite4Unity3d.SQLiteConnection.Delete[T] (System.Object primaryKey) (at Assets/CleverTap/Runtime/Native/UnityNativeWrapper/Sqlite/Scripts/UnitySQLite.cs:1512)
CleverTapSDK.Native.SQLiteDataService.Delete[T] (System.Int32 id) (at Assets/CleverTap/Runtime/Native/UnityNativeWrapper/Sqlite/Scripts/SQLiteDataService.cs:34)
CleverTapSDK.Native.UnityNativeDatabaseStore.DeleteEvents (System.Collections.Generic.List`1[T] eventsToRemove) (at Assets/CleverTap/Runtime/Native/UnityNativeWrapper/UnityNativeDatabaseStore.cs:73)
```

- Handshake `/hello` results in a ProtocolError, status code is 400 (running the app in the Unity Editor)

- The following error is still logged in the native binding.
```Start: no-op CleverTap binding for non iOS/Android.```

**Pull request closed without merging**